api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,278 @@
+    return context.SetValue(span_key, sp);
+  }
+
+  static void GetCurrentSpan(const context::Context &context, SpanContext &span_context)
+  {
+    const nostd::string_view span_key = "current-span";
+    context::Context ctx(context);
+    context::ContextValue span = ctx.GetValue(span_key);
+    if (nostd::holds_alternative<nostd::shared_ptr<Span>>(span))
+    {
+      span_context = nostd::get<nostd::shared_ptr<Span>>(span).get()->GetContext();
+    }
+  }
+
+  static TraceId GenerateTraceIdFromString(nostd::string_view trace_id)
+  {
+    const char *trc_id = trace_id.begin();
+    uint8_t buf[kTraceIdBytes / 2];
+    for (int i = 0; i < kTraceIdBytes; i++)
+    {
+      int tmp = CharToInt(*trc_id);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kTraceIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return TraceId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+      trc_id++;
+    }
+    return TraceId(buf);
+  }
+
+  static SpanId GenerateSpanIdFromString(nostd::string_view span_id)
+  {
+    const char *spn_id = span_id.begin();
+    uint8_t buf[kSpanIdBytes / 2];
+    for (int i = 0; i < kSpanIdBytes; i++)
+    {
+      int tmp = CharToInt(spn_id[i]);
+      if (tmp < 0)
+      {
+        for (int j = 0; j < kSpanIdBytes / 2; j++)
+        {
+          buf[j] = 0;
+        }
+        return SpanId(buf);
+      }
+      if (i % 2 == 0)
+      {
+        buf[i / 2] = tmp * 16;
+      }
+      else
+      {
+        buf[i / 2] += tmp;
+      }
+    }
+    return SpanId(buf);
+  }
+
+  static TraceFlags GenerateTraceFlagsFromString(nostd::string_view trace_flags)
+  {
+    int tmp1 = CharToInt(trace_flags[0]);
+    int tmp2 = CharToInt(trace_flags[1]);
+    if (tmp1 < 0 || tmp2 < 0)
+      return TraceFlags(0);  // check for invalid char
+    uint8_t buf = tmp1 * 16 + tmp2;
+    return TraceFlags(buf);
+  }
+
+private:
+  // Converts a single character to a corresponding integer (e.g. '1' to 1), return -1
+  // if the character is not a valid number in hex.
+  static uint8_t CharToInt(char c)
+  {
+    if (c >= '0' && c <= '9')
+    {
+      return (int)(c - '0');
+    }
+    else if (c >= 'a' && c <= 'f')
+    {
+      return (int)(c - 'a' + 10);
+    }
+    else if (c >= 'A' && c <= 'F')
+    {
+      return (int)(c - 'A' + 10);
+    }
+    else
+    {
+      return -1;
+    }
+  }
+
+  static void InjectTraceParent(const SpanContext &span_context, T &carrier, Setter setter)
+  {
+    char trace_id[32];
+    TraceId(span_context.trace_id()).ToLowerBase16(trace_id);
+    char span_id[16];
+    SpanId(span_context.span_id()).ToLowerBase16(span_id);
+    char trace_flags[2];
+    TraceFlags(span_context.trace_flags()).ToLowerBase16(trace_flags);
+    // Note: This is only temporary replacement for appendable string
+    std::string hex_string = "00-";
+    for (int i = 0; i < 32; i++)
```

Comment:
Feel free to ignore my comment if OpenTelemetry doesn't allow this, but you can use `std::string::append`.

Reply:
I feel that the addition feels more intuitive than invoking append(), that's the reason I am using it

Reply:
I've changed them to append